### PR TITLE
Api handler

### PR DESCRIPTION
**Pull request by egedebruin**
This handles http requests to set a chest to be opened. #58 

When the video is running, the server is started and you can set that the next chest is opened by going to the url:

localhost:[portnumber]/chest?opened=true

With [portnumber] = 8080 - or if you want to change it, you can change the default port number in APIHandler. I've also created a method which can change the port number, by for example a number set in the configuration file. I don't know if this would be useful(l).

useful or not, I added the possibility.

When the url request is done, the next chest is set opened which is shown in the progressbar and in the log.

### CODE REVIEW

File: src/main/java/api/APIChestHandler.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     */
+    public APIChestHandler(final MainController mainController) {
+        controller = mainController;
+    }
+
+
+    @Override
+    public void handle(final String s, final Request request, final HttpServletRequest httpRequest,
+                       final HttpServletResponse response) throws IOException, ServletException {
+        String body = "Ajax altijd nümmer 1";
```
Comment (lisv) [inline src/main/java/api/APIChestHandler.java]: wat

Comment (egedebruin) [reply src/main/java/api/APIChestHandler.java]: (:

**Comment by egedebruin** [comment]
ok thx for de info

File: src/main/java/api/APIChestHandler.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * Constructor for the APIChestHandler.
+     * @param mainController the mainController for this handler.
+     */
+    public APIChestHandler(final MainController mainController) {
+        controller = mainController;
+    }
+
+
```
Comment (lisv) [inline src/main/java/api/APIChestHandler.java]: Two spacebars

Comment (egedebruin) [reply src/main/java/api/APIChestHandler.java]: spacebars?

Comment (lisv) [reply src/main/java/api/APIChestHandler.java]: Two enters I mean, two empty lines

File: src/main/java/api/APIChestHandler.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    @Override
+    public void handle(final String s, final Request request, final HttpServletRequest httpRequest,
+                       final HttpServletResponse response) throws IOException, ServletException {
```
Comment (lisv) [inline src/main/java/api/APIChestHandler.java]: This exception is never thrown (ServletException)

**Pull request merged**